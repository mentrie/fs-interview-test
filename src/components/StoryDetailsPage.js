import React from 'react'


const StoryDetailsPage = ({story}) => {
  return (
    <div>
      <header>Story Details Page</header>
      <h4>{story.title}</h4>
      <div>
        <p>Score: {story.score}</p>
        <p>Comments: {story.descendants}</p>
        <p>Author: {story.by}</p>
        <p>Website: {story.url}</p>
      </div>
    </div>
  )
}

export default StoryDetailsPage;