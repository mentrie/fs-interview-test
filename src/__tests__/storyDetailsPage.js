import React from 'react';
import renderer from 'react-test-renderer';
import StoryDetailsPage from '../components/StoryDetailsPage';

it('Should render component with empty props',  () => {
  const component = renderer.create(<StoryDetailsPage story={{}} />);

  expect(component.toJSON()).toMatchSnapshot();
})

it('Should render component with props provided', () => {
  const component = renderer.create(<StoryDetailsPage story={{title: 'Testing', id: 1}} />);
  expect(component.toJSON()).toMatchSnapshot();
});

