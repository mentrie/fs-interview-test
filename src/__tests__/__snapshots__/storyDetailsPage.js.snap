// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component with empty props 1`] = `
<div>
  <header>
    Story Details Page
  </header>
  <h4 />
  <div>
    <p>
      Score: 
    </p>
    <p>
      Comments: 
    </p>
    <p>
      Author: 
    </p>
    <p>
      Website: 
    </p>
  </div>
</div>
`;

exports[`Should render component with props provided 1`] = `
<div>
  <header>
    Story Details Page
  </header>
  <h4>
    Testing
  </h4>
  <div>
    <p>
      Score: 
    </p>
    <p>
      Comments: 
    </p>
    <p>
      Author: 
    </p>
    <p>
      Website: 
    </p>
  </div>
</div>
`;
